kalliope/core/OrderAnalyser.py

```diff
@@ -71,6 +71,7 @@ def get_list_synapses_to_process(cls, list_match_synapse):
     @classmethod
     def get_list_match_synapse(cls, order, synapse_order_tuple):
```

Comment:
Note : Not part of this PR, but this method `get_list_match_synapse` starts to be way too complicated. Need to be refactored and simplified. Will create an issue for that

Reply:
Add unittests as well

Reply:
Thanks for tests!